src/dune_rules/lib.ml

```diff
@@ -1207,6 +1224,16 @@ end = struct
     in
     let open Memo.O in
     let open Resolved_deps_builder in
+    let deps =
+      let ocaml_version = db.lib_config.ocaml_version in
+      let bigarray_in_std_libraries =
+        Ocaml.Version.has_bigarray_library ocaml_version
+      in
+      if bigarray_in_std_libraries then deps
+      else
+        let bigarray = Lib_name.of_string "bigarray" in
```

Comment:
This change is incorrect and introduces a bug that forbids local libraries named `bigarray` from being used on OCaml < 5.0.

Reply:
True, I don't know how we can test if there a local library called `bigarray`.

Reply:
that has been fixed in #8902 .